src/Enchantments.h

```diff
@@ -103,6 +104,9 @@ class cEnchantments
 	/** Returns true if there are no enchantments */
 	bool IsEmpty(void) const;
 
+	/** Returns true if all of our enchantments are compatible with the specified enchantment */
+	bool IsCompatibleWith(int a_EnchantmentID) const;
```

Comment:
For thought: how about `CanAddEnchantment`, at least for me that is a more descriptive name.